tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/Tests/ActionTests/Action_Foreach_Updated.test.dialog

```diff
@@ -0,0 +1,56 @@
+{
+    "$schema": "../../../tests.schema",
+    "$kind": "Microsoft.Test.Script",
+    "dialog": {
+        "$kind": "Microsoft.AdaptiveDialog",
+        "id": "root",
+        "triggers": [
+            {
+                "$kind": "Microsoft.OnUnknownIntent",
+                "actions": [
+                    {
+                        "$kind": "Microsoft.SetProperty",
+                        "property": "dialog.todo",
+                        "value": "=['a', 'b', 'c']"
+                    },
+                    {
+                        "$kind": "Microsoft.Foreach",
+                        "itemsProperty": "dialog.todo",
+                        "actions": [
+                            {
+                                "$kind": "Microsoft.SetProperty",
```

Comment:
I don't think this is the right behavior.  you have cloned the array, but that means that you can't modify the contents of the array.  I don't think the semantics of this is right.

Reply:
I just want to make sure that any update on the itemsProperty will not affect the foreach loop. which means, even you have updated the itemsProperty within the loop, it will still loop the original items. see #3901 for more details.

My PR here is not complete and ready to review,  could you see my previous comments and give suggestions on that?

Reply:
I would not do it this way because you are making it that NO modifications can be made to items in the collection which defeats the purpose of the action.  

Why are you trying to prevent this?  If people remove the items or change the collection it's their own dang fault.  

---
In reply to: [448080768](https://github.com/microsoft/botbuilder-dotnet/pull/3904#discussion_r448080768) [](ancestors = 448080768)

Reply:
@tomlm, this is all because of this issue #3901 , I am fine to not fix this and just let user to set the turn.activities to other properties and then do the loop. Could you comment on that issue?